pilot/pkg/bootstrap/server.go

```diff
@@ -272,25 +275,25 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 		return nil, err
 	}
 
-	if err := s.initControllers(args); err != nil {
+	// Parse and validate Istiod Address.
+	istiodHost, _, err := e.GetDiscoveryAddress()
+	if err != nil {
 		return nil, err
 	}
 
-	s.XDSServer.InitGenerators(e, args.Namespace, s.internalDebugMux)
-
-	// Initialize workloadTrustBundle after CA has been initialized
-	if err := s.initWorkloadTrustBundle(args); err != nil {
+	// Create Istiod certs and setup watches.
+	if err := s.initIstiodCerts(args, string(istiodHost)); err != nil {
 		return nil, err
 	}
 
-	// Parse and validate Istiod Address.
-	istiodHost, _, err := e.GetDiscoveryAddress()
-	if err != nil {
+	if err := s.initControllers(args); err != nil {
 		return nil, err
 	}
 
-	// Create Istiod certs and setup watches.
-	if err := s.initIstiodCerts(args, string(istiodHost)); err != nil {
+	s.XDSServer.InitGenerators(e, args.Namespace, s.internalDebugMux)
+
+	// Initialize workloadTrustBundle after CA has been initialized
+	if err := s.initWorkloadTrustBundle(args); err != nil {
```

Comment:
Is the order here significant?

Reply:
What do you suggest? Do you think the current order is not correct?

Reply:
It is different than what the current upstream has so I wonder if there's a significance here. But I lack the knowledge if that was a deliberate decision. Clearly, tests work both ways, so from this limited viewpoint, it's irrelevant.

Reply:
Anything from the top of your head @dgn @luksa?

Reply:
No I think the changes above this line are not needed. Only the one that adds the federation struct

Reply:
Ok I will restore the order so it might result in less merge work in the future.

Reply:
I wonder however if we don't need `s.initIstiodCerts` call first.